File: Leetcode/190. Reverse Bits/reverse_bits.py
#https://leetcode.com/problems/single-number-ii/solutions/3714928/bit-manipulation-c-java-python-beginner-friendly/

class Solution:
    def reverseBits(self, n: int) -> int:
        ans = 0
        for i in range(32):
            last_bit = n>>i&1
            ans = ans | last_bit<<(31-i)
        return ans


item  = Solution()
print(item.reverseBits(43261596))

File: Leetcode/190. Reverse Bits/reverse_bits.rs
struct Solution{}

impl Solution {
    pub fn reverse_bits(x: u32) -> u32 {
        let mut ans =0;
        for i in 0..32u32{
            let last_bit = x>>i&1;
            ans = ans |(last_bit<<(31-i))
        }
        ans
        
    }
}
fn main() {
    let res = Solution::reverse_bits(43261596);
    println!("{}",res)
}
